src/lib/es5.d.ts

```diff
@@ -4409,12 +4409,25 @@ declare namespace Intl {
 
     var Collator: CollatorConstructor;
 
+    interface NumberFormatOptionsStyleRegistry {
+        decimal: any;
+        percent: any;
+        currency: any;
+    }
+
+    type NumberFormatOptionsStyle = keyof NumberFormatOptionsStyleRegistry;
+
+    interface NumberFormatOptionsUseGroupingRegistry {}
+
+    type NumberFormatOptionsUseGrouping = {} extends NumberFormatOptionsUseGroupingRegistry ? boolean : keyof NumberFormatOptionsUseGroupingRegistry | "true" | "false" | boolean;
+    type ResolvedNumberFormatOptionsUseGrouping = {} extends NumberFormatOptionsUseGroupingRegistry ? boolean : keyof NumberFormatOptionsUseGrouping;

```

Comment:
This seems weird; why not just always allow boolean?

Reply:
The `Resolved` types are returned from the API to the user, as opposed to the option types which are input from the user to the API. For most cases, these are identical, but useGrouping is different since es2023.

In the initial spec, both the input option and the resolved option returned were both of type boolean.
With the 2023 definition for the useGrouping option, the library accepts a boolean (or "true" | "false") as input, but will only ever return a member of the string union as a resolved option value.

This is more correct, at the expense of introducing a conditional type into the library (although there are other examples of conditional use, eg. es2019.array)